docs/194-inicjalizacja.md

```diff
@@ -0,0 +1,87 @@
+# Inicjalizacja bota
+  
+## Punkt wejściowy
+
+Jeśli uruchamiamy bota lokalnie, punktem wejściowym jest plik  "Program" w projekcie "Watchman.Discord". Tutaj
+uruchamiana jest tylko czysta aplikacja samego bota, czyli tworzony i konfigurowany obiekt klasy "WatchmanBot".
+Na produkcji bot jest uruchamiany w tle przez aplikację webową (projekt "Watchman.Web") i obsługiwany przez Hangfire.
+Obiekt klasy "WatchmanBot" jest tworzony w pliku "Watchman.Web/ServiceProviders/AutofacServiceProviderFactory" w funkcji:
+```
+public IServiceProvider CreateServiceProvider(ContainerBuilder containerBuilder)
+```
+Również w tym miejscu, po stworzeniu bota konfigurowane będą serwisy pracujące w tle, wymagane do jego obsługi:
+```
+container.Resolve<HangfireJobsService>().SetDefaultJobs(container)
+```
+
+## InitializationService
+
+```
+    public async Task InitServer(DiscordServerContext server)
+    { 
+        Log.Information("Initializing server: {server}", server.ToJson());
+        await this.MuteRoleInit(server);
+        var lastInitDate = this.GetLastInitDate(server);
+        await this.ReadServerMessagesHistory(server, lastInitDate);
+        await this._cyclicStatisticsGeneratorService.GenerateStatsForDaysBefore(server, lastInitDate);
+        await this.NotifyDomainAboutInit(server);
+        Log.Information("Done server: {server}", server.ToJson());
+    }
+```
+InitializationService jest używany w klasie "WatchmanBot" w funkcji "GetWorkflowBuilder()".
+Metoda "InitServer" zajmuje się wszystkim co potrzebne aby dostać na podanym w argumencie serwerze gotowego do pracy bota.
```

Comment:
widziałem tą dokumentacje bardziej jako liste elementów które dzieją się podczas inicjalizacji 
czyli
-robimy to i to ze statystykami
-robimy to i to z wyciszonymi użytkownikami
-itd

i to jest element na którym bym się skupił 
reszte potraktowałbym jako dodatek, bo sam proces inicjalizacji jest dość prosty (odpalamy ją i sie dzieje)
mimo to warto go opisać -> im więcej mamy informacji w dokumentacji tym lepiej

ta strona w większości sytuacji będzie używana jako wyjaśnienie `dlaczego to się stało`
przykładowo `dlaczego w bazie pojawiła sie taka lista`
albo `dlaczego system wie o tym`

Reply:
jest lepiej teraz?